Tests/Unit/NetworkServer/LoRaDeviceTest.cs

```diff
@@ -876,5 +876,144 @@ public async Task When_Initialized_With_Class_C_And_Custom_RX2DR_Should_Have_Cor
+            }
+
+            [Theory]
+            [InlineData(10, 10, 0, 0, true)]
+            [InlineData(10, 10, 10, 10, false)]
+            public void When_Reset_Specified_Initialized_Correctly(uint startDesiredUp, uint startDesiredDown, uint startReportedUp, uint startReportedDown, bool expectStart)
+            {
+                using var device = CreateDefault();
+
+                const uint fcntUp = 10;
+                const uint fcntDown = 2;
+                var twin = TestUtils.CreateTwin(desired: new Dictionary<string, object>
+                                                {
+                                                    [TwinProperty.FCntUpStart] = startDesiredUp,
+                                                    [TwinProperty.FCntDownStart] = startDesiredDown
+                                                },
+                                                reported: new Dictionary<string, object>
+                                                {
+                                                    [TwinProperty.FCntUpStart] = startReportedUp,
+                                                    [TwinProperty.FCntDownStart] = startReportedDown,
+                                                    [TwinProperty.FCntUp] = fcntUp,
+                                                    [TwinProperty.FCntDown] = fcntDown
+                                                });
+
+                device.ExecuteInitializeFrameCounters(twin);
+
+                this.loRaDeviceClient.Verify(x => x.UpdateReportedPropertiesAsync(It.IsAny<TwinCollection>()), expectStart ? Times.Once : Times.Never);
+
+                if (expectStart)
+                {
+                    AssertFcntUp(startDesiredUp, device);
+                    AssertFcntDown(startDesiredDown, device);
+                }
+                else
+                {
+                    AssertFcntUp(fcntUp, device);
+                    AssertFcntDown(fcntDown, device);
+                }
+            }
+
+            private static void AssertFcntUp(uint expected, LoRaDevice device)
+                => AssertEqual(expected, new[] { device.FCntUp, device.LastSavedFCntUp });
+            
+            private static void AssertFcntDown(uint expected, LoRaDevice device)
+                => AssertEqual(expected, new[] { device.FCntDown, device.LastSavedFCntDown });
+
+            private static void AssertEqual<T>(T expected, IEnumerable<T> it)
+                => it.ForEach(x => Assert.Equal(expected, x));
+
+            private LoRaDeviceTest CreateDefault()
+                => new LoRaDeviceTest(this.loRaDeviceClient.Object);
+
+            private class LoRaDeviceTest : LoRaDevice
+            {
+                private readonly ILogger<TwinCollectionReader> logger = NullLogger<TwinCollectionReader>.Instance;
+
+                public LoRaDeviceTest(ILoRaDeviceClient deviceClient)
+#pragma warning disable CA2000 // Dispose objects before losing scope
```

Comment:
[nit] maybe a comment that ownership is transferred to the `LoRaDevice` would be helpful, then it's not necessary to look through the code to figure out if it's a bug that it's suppressed.

Side note: apparently we just `Release()` the `connectionManager` inside the loradevice, but we don't `Dispose()` it on disposal. Do we need to change this? (It should execute the same logic as of now, but that might change in the future, no?)

Reply:
I'll add the comment. I thought it was obvious, as we do not keep a reference anywhere and pass it into the ctor of the item. It's a typical DI pattern. 

The connection manager is a singleton that is managing single device connections. We do not dispose it unless we are shutting down the app. We only release the connection that is managed for that particular device. There are a few issues with the pattern as it is implemented atm, and all of those are tracked as part of the caching improvement feature.

Reply:
0e5a8b092caca1617ffa0117132f6e61f1e9f0c0